frontend/views/containers/dashboard/GroupMembers.vue

```diff
@@ -54,45 +42,64 @@
 import { mapGetters } from 'vuex'
 import { OPEN_MODAL } from '@utils/events.js'
 import sbp from '~/shared/sbp.js'
+import Avatar from '@components/Avatar.vue'
 import AvatarUser from '@components/AvatarUser.vue'
-import GroupMemberMenu from '@containers/dashboard/GroupMemberMenu.vue'
-import Tooltip from '@components/Tooltip.vue'
+import GroupMembersMenu from '@containers/dashboard/GroupMembersMenu.vue'
+import GroupMembersTooltipPending from '@containers/dashboard/GroupMembersTooltipPending.vue'
 
 export default {
   name: 'GroupMembers',
   components: {
+    Avatar,
     AvatarUser,
-    GroupMemberMenu,
-    Tooltip
-  },
-  methods: {
-    invite () {
-      this.$router.push({ path: '/invite' })
-    },
-    openModal (modal, queries) {
-      sbp('okTurtles.events/emit', OPEN_MODAL, modal, queries)
-    }
+    GroupMembersMenu,
+    GroupMembersTooltipPending
   },
   computed: {
     ...mapGetters([
       'groupProfiles',
       'groupShouldPropose',
       'groupMembersCount',
-      'ourUsername'
+      'ourUsername',
+      'userDisplayName',
+      'currentGroupState',
+      'groupMembersPending'
     ]),
+    weJoined () {
+      return this.currentGroupState.profiles[this.ourUsername].joined_ms
+    },
     firstTenMembers () {
       const profiles = this.groupProfiles
-      const usernames = Object.keys(profiles).slice(0, 10)
-      return usernames.reduce((acc, username) => {
+      const sliceIndex = 10 - Math.min(10, Object.keys(this.groupMembersPending).length) // avoid slicing too many members.
+      const usernames = Object.keys(profiles).slice(0, sliceIndex)
+      const members = usernames.reduce((acc, username) => {
         // Prevent displaying users without a synced contract.
         // It happens at the exact moment a user joins a group and both
         // contracts (group + user) are still syncing
         const profile = profiles[username]
         if (profile) {
-          acc[username] = profile
+          acc[username] = { profile }
         }
         return acc
       }, {})
+
+      // TODO - Sort members #869
+      return { ...this.groupMembersPending, ...members }
+    }
+  },
+  methods: {
+    invite () {
+      this.$router.push({ path: '/invite' })
+    },
+    openModal (modal, queries) {
+      sbp('okTurtles.events/emit', OPEN_MODAL, modal, queries)
+    },
+    isNewMember (username) {
+      if (username === this.ourUsername) { return false }
+
+      const memberJoined = this.currentGroupState.profiles[username].joined_ms
```

Comment:
This needs to be converted to milliseconds

```js
const joined_ms = new Date(this.currentGroupState.profiles[username].joinedDate).getTime()
```

Reply:
Changed where needed